src/algorithms/cutoff.rs

```diff
@@ -71,11 +71,11 @@ impl Guesser for Cutoff {
                 // considering a world where we _did_ guess `word` and got `pattern` as the
                 // correctness. now, compute what _then_ is left.
                 let mut in_pattern_total = 0;
+                let g = Guess {
+                    word: Cow::Borrowed(word),
+                    mask: *pattern,
+                };
                 for (candidate, count) in &*self.remaining {
-                    let g = Guess {
-                        word: Cow::Borrowed(word),
-                        mask: *pattern,
-                    };
```

Comment:
> That suggests we should break out #[inline] and #[inline(always)] to see if we can claw some of that back.

I think creating the `Guess` inside the loop might have been the difference. My initial version didn't create one, and moving it outside the loop gives the same speed.